build-tools/packages/build-tools/src/common/tsCompile.ts

```diff
@@ -40,31 +70,60 @@ export function tsCompile({
 
 	let code = ts.ExitStatus.DiagnosticsPresent_OutputsSkipped;
 	if (commandLine && diagnostics.length === 0) {
+		// When specified overrides current directory's package.json type field so tsc may cleanly
+		// transpile .ts files to CommonJS or ESM using compilerOptions.module Node16 or NodeNext.
+		let packageJsonTypeOverrideUsage = "not read" as "not read" | "already present" | "used";
+		const applyPackageJsonTypeOverride = !packageJsonTypeOverride
+			? undefined
+			: (
+					host:
+						| tsTypes.CompilerHost
+						| tsTypes.WatchCompilerHostOfConfigFile<tsTypes.EmitAndSemanticDiagnosticsBuilderProgram>,
+				): void => {
+					const origReadFile = host.readFile;
```

Comment:
Nit: `originalReadFile`?

Reply:
Sure - this was relocated code, but always nice to incrementally improve.